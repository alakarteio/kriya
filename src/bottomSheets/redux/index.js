import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
