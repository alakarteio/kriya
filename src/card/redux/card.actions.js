export const OPEN = 'CARD_OPEN'
export const open = name => ({ type: OPEN, payload: name })

export const CLOSE = 'CARD_CLOSE'
export const close = name => ({ type: CLOSE, payload: name })

export const TOGGLE = 'CARD_TOGGLE'
export const toggle = name => ({ type: TOGGLE, payload: name })

export const CLOSE_ALL = 'CARD_CLOSE_ALL'
export const closeAll = () => ({ type: CLOSE_ALL })

export const OPEN_ALL = 'CARD_OPEN_ALL'
export const openAll = () => ({ type: OPEN_ALL })
