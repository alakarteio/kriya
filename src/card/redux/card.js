import {
  OPEN,
  CLOSE,
  TOGGLE,
  OPEN_ALL,
  CLOSE_ALL,
} from './card.actions'

export const initState = {}
export const initAction = { type: 'UNKOWN' }

export default (state = initState, { type, payload } = initAction) => {
  switch (type) {
    case OPEN: return { ...state, [payload]: true }
    case CLOSE: return { ...state, [payload]: false }
    case TOGGLE: return { ...state, [payload]: !state[payload] }
    case OPEN_ALL:
      return Object.keys(state)
        .reduce(
          (acc, next) => ({ ...acc, [next]: true }),
          {},
        )
    case CLOSE_ALL:
      return Object.keys(state)
        .reduce(
          (acc, next) => ({ ...acc, [next]: false }),
          {},
        )
    default: return state
  }
}
