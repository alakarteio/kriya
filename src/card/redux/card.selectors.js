export const getCards = state => state.ui.cards
export const getCard = state => id => getCards(state)[id]

export const isOpened = state => id => getCard(state)(id)
