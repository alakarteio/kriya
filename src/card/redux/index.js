export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
