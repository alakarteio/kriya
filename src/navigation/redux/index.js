export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
