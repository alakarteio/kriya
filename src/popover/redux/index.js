export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
