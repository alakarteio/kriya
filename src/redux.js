export { default as cards } from './card/redux'
export { default as navigation } from './navigation/redux'
export { default as popover } from './popover/redux'
export { default as bottomSheets } from './bottomSheets/redux'
export { default as tabs } from './tabs/redux'
