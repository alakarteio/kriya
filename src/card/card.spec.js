/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { open } from './redux'
import Card from './card'
import CardContainer from './card.container'

describe('common/Card', () => {
  describe('graphical (JSX)', () => {
    const snap = (props, state = {}) => {
      const store = createStore(() => (state))
      /* eslint-disable react/no-children-prop */
      const component = renderer.create(
        <Provider store={store}>
          <Card {...props} />
        </Provider>,
      )
      /* eslint-enable react/no-children-prop */

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    }

    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should have a default behaviour', () => snap({}))
    it('should print the closed element', () => snap({ closed: true, closeElm: 'close elm', children: 'children' }))
    it('should print the children', () => snap({ closed: false, closeElm: 'close elm', children: 'children' }))
  })

  describe('container', () => {
    const snap = (props, cards = {}) => {
      const dispatch = jest.fn()
      const store = createStore(() => ({ ui: { cards } }))
      store.dispatch = dispatch

      const wrapper = mount(
        <Provider store={store}>
          <CardContainer {...props} />
        </Provider>,
      )

      wrapper.find('.card').simulate('click')

      return dispatch
    }

    it('should call the given click', () => {
      let onClick = jest.fn()
      snap({ name: 'card1', onClick }, { card1: false })
      expect(onClick.mock.calls.length).toBe(1)

      onClick = jest.fn()
      snap({ name: 'card1', onClick }, { card1: true })
      expect(onClick.mock.calls.length).toBe(1)
    })

    it('should open the card', () => {
      const dispatch = snap({ name: 'card1' }, { card1: false })

      expect(dispatch.mock.calls.length).toBe(1)
      expect(dispatch.mock.calls[0]).toEqual([open('card1')])
    })

    it('should not close the card', () => {
      const dispatch = snap({ name: 'card1' }, { card1: true })

      expect(dispatch.mock.calls.length).toBe(0)
    })
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
