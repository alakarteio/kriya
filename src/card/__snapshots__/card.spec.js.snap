// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Card graphical (JSX) should add custom className 1`] = `
<div
  className="card custom"
  style={Object {}}
/>
`;

exports[`common/Card graphical (JSX) should add custom style 1`] = `
<div
  className="card"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;

exports[`common/Card graphical (JSX) should have a default behaviour 1`] = `
<div
  className="card"
  style={Object {}}
/>
`;

exports[`common/Card graphical (JSX) should print the children 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div>
    Children 
    {}
  </div>
</div>
`;

exports[`common/Card graphical (JSX) should print the children with a closed prop 1`] = `
<div
  className="card closed"
  style={Object {}}
>
  <div>
    Children 
    {"closed":true}
  </div>
</div>
`;

exports[`common/Card graphical (JSX) should print the closed element 1`] = `
<div
  className="card closed"
  style={Object {}}
>
  close elm
</div>
`;
